File: code/labo16/labo16-client/app/_types/dinosaur.ts
import { Incident } from "./incident";
import { Zookeeper } from "./zookeeper";

export class Dinosaur{
    constructor(
        public id : number,
        public name : string,
        public specie : string,
        public incidents : Incident[],
        public zookeeper : Zookeeper
    ){}
}

File: code/labo8.2/labo16_serveur/Models/Dinosaur.cs
namespace serveur15.Models
{
    public class Dinosaur
    {
        public int Id { get; set; }
        public string Name { get; set; } = null!;
        public string Specie { get; set; } = null!;
        public virtual List<Incident> Incidents { get; set; } = new List<Incident>();
        public virtual Zookeeper Zookeeper { get; set; } = null!;
    }
}
